CSharp/TextSearch/Common/Helpers.cs

```diff
@@ -46,67 +42,63 @@ public static string GetSplitFileName(int fileNumber)
+                    Console.WriteLine($"Failure Message: {failureInformation.Message}");
+                    Console.WriteLine($"Failure Category: {failureInformation.Category}");
+                    Console.WriteLine("Failure Details:");
 
-                        throw new TextSearchException(String.Format("Task {0} failed with a scheduling error", boundTask.Id));
+                    foreach (NameValuePair detail in failureInformation.Details)
+                    {
+                        Console.WriteLine("{0} : {1}", detail.Name, detail.Value);
                     }
-                    
-                    //Read the content of the output files if the task exited.
+
                     if (boundTask.ExecutionInformation.ExitCode.HasValue)
                     {
-                        Console.WriteLine("Task {0} exit code: {1}", boundTask.Id, boundTask.ExecutionInformation.ExitCode);
+                        Console.WriteLine($"Task {boundTask.Id} exit code: {boundTask.ExecutionInformation.ExitCode}");
 
-                        if (dumpStandardOutOnTaskSuccess && boundTask.ExecutionInformation.ExitCode.Value == 0 || boundTask.ExecutionInformation.ExitCode.Value != 0)
+                        if (boundTask.ExecutionInformation.ExitCode.Value != 0)
                         {
-                            //Dump the standard out file of the task.
-                            NodeFile taskStandardOut = await boundTask.GetNodeFileAsync(Batch.Constants.StandardOutFileName);
-
-                            Console.WriteLine("Task {0} StdOut:", boundTask.Id);
-                            Console.WriteLine("----------------------------------------");
-                            string stdOutString = await taskStandardOut.ReadAsStringAsync();
-                            result = stdOutString;
-                            Console.WriteLine(stdOutString);
+                            await GetFileAsync(boundTask, Batch.Constants.StandardOutFileName);
                         }
 
-                        //Check for nonzero exit code and dump standard error if there was a nonzero exit code.
                         if (boundTask.ExecutionInformation.ExitCode.Value != 0)
                         {
-                            NodeFile taskErrorFile = await boundTask.GetNodeFileAsync(Batch.Constants.StandardErrorFileName);
-
-                            Console.WriteLine("Task {0} StdErr:", boundTask.Id);
-                            Console.WriteLine("----------------------------------------");
-                            string stdErrString = await taskErrorFile.ReadAsStringAsync();
-                            Console.WriteLine(stdErrString);
-
-                            throw new TextSearchException(String.Format("Task {0} failed with a nonzero exit code", boundTask.Id));
+                            await GetFileAsync(boundTask, Batch.Constants.StandardErrorFileName);
```

Comment:
Do we still need two separate if-guards around these requests?  Seems like they are now the same condition.